lib/MusicBrainz/LWP/TimelyUserAgent.pm

```diff
@@ -0,0 +1,110 @@
+package MusicBrainz::LWP::TimelyUserAgent;
+
+use Moose;
+
+use DBDefs;
+use HTTP::Status qw( HTTP_SERVICE_UNAVAILABLE );
+use MusicBrainz::Server::Exceptions qw( GenericTimeout );
+use POSIX qw( SIGALRM );
+use Try::Tiny;
+
+extends 'LWP::UserAgent';
+
+use MooseX::Types::Moose qw( Int );
+
+has 'sending_timeout' => (
+    is => 'rw',
+    isa => 'Int',
+);
+
+sub new {
+    my ($class, %args) = @_;
+
+    $args{agent} = DBDefs->LWP_USER_AGENT unless exists $args{agent};
+
+    my $sending_timeout =
+        exists $args{sending_timeout}
+            ? delete $args{sending_timeout}
+            : 1;
+
+    my $obj = $class->SUPER::new(%args);
+
+    return $class->meta->new_object(
+        __INSTANCE__ => $obj,
+        sending_timeout => $sending_timeout,
+    );
```

Comment:
https://metacpan.org/pod/release/FLORA/Moose-1.01/lib/Moose/Cookbook/Basics/Recipe11.pod for anyone else like me who was unfamiliar with this pattern